packages/core/components/DirectoryTree/findChildNodes.ts

```diff
@@ -0,0 +1,100 @@
+import { defaults, isEmpty, pull } from "lodash";
+
+import { NO_VALUE_NODE, ROOT_NODE } from "./directory-hierarchy-state";
+import FileSet from "../../entity/FileSet";
+import ExcludeFilter from "../../entity/FileFilter/ExcludeFilter";
+import { AnnotationService, FileService } from "../../services";
+import { naturalComparator } from "../../util/strings";
+
+export interface FindChildNodesParams {
+    ancestorNodes?: string[];
+    currentNode: string;
+    hierarchy: string[];
+    annotationService: AnnotationService;
+    fileService: FileService;
+    fileSet: FileSet;
+    shouldShowNullGroups?: boolean;
+}
+
+const DEFAULTS = {
+    ancestorNodes: [],
+    currentNode: ROOT_NODE,
+    shouldShowNullGroups: false,
+};
+
+export async function findChildNodes(params: FindChildNodesParams): Promise<string[]> {
+    const {
+        ancestorNodes,
+        currentNode,
+        fileSet,
+        hierarchy,
+        annotationService,
+        fileService,
+        shouldShowNullGroups,
+    } = defaults({}, params, DEFAULTS);
+
+    const isRoot = currentNode === ROOT_NODE;
+    // If at root of hierarchy, currentNode will be set to "ROOT_NODE"
+    // We trim that from the path as it is not meaningful in this context
+    const pathToNode = pull([...ancestorNodes, currentNode], ROOT_NODE);
+
+    let values: string[] = [];
+
+    const depth = pathToNode.length;
+    const annotationNameAtDepth = hierarchy[depth];
+    if (fileSet.excludeFilters?.some((filter) => filter.name === annotationNameAtDepth)) {
+        // User does not want files with this annotation; don't return any non-null values.
+        return shouldShowNullGroups ? [NO_VALUE_NODE] : [];
```

Comment:
We need to do a check for the count here of the `NO_VALUE_NODE` right to make sure it has values?

Reply:
ah true, I'll move the count up to the top

Reply:
fixed in [d01d0fb](https://github.com/AllenInstitute/biofile-finder/pull/498/commits/d01d0fb644e756e7f00aa092a4e9b2062af85151)